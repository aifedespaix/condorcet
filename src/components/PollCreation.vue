<template>
  <div>
    <b-field label="Nom"> <b-input v-model="name"></b-input> </b-field>
    <b-field label="Description">
      <b-input v-model="description" maxlength="200" type="textarea"></b-input>
    </b-field>
    <b-field label="Ajouter des réponses">
      <b-taginput
        v-model="answers"
        ellipsis
        icon="label"
        placeholder="Ajouter une réponse"
      >
      </b-taginput>
    </b-field>
    <b-field>
      <button class="button" @click="createPoll">Créer</button>
    </b-field>
  </div>
</template>

<script>
import { mapState, mapActions } from "vuex";
import randomString from "random-string";
import { db } from "@/firebase";

export default {
  name: "PollCreation",
  data() {
    return {
      name: "",
      description: "",
      answers: []
    };
  },
  computed: {
    ...mapState({
      userId: state => state.user.id
    })
  },
  methods: {
    ...mapActions("app", ["addNotification"]),
    async createPoll() {
      try {
        let users = [this.userId];

        await db.collection("polls").add({
          name: this.name,
          description: this.description,
          answers: this.answers,
          owner: this.userId,
          is_active: true,
          token: randomString({ length: 20 }),
          // timestamp: db.FieldValue.serverTimestamp,
          users
        });

        this.addNotification({
          text: "Scrutin créé mon pote",
          status: "success"
        });
      } catch (error) {
        this.addNotification({
          text: `Il y a eu un problème lors de la création du scrutin`,
          status: "error"
        });
      }
    },
    searchEmail() {
      this.$firestoreRefs.polls;
    }
  }
};
</script>

<style></style>
