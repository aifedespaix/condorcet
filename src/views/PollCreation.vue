<template>
  <div>
    <v-text-field label="Nom" v-model="name"></v-text-field>

    <v-textarea
      name="input-7-1"
      label="Description"
      v-model="description"
    ></v-textarea>

    <v-text-field
      label="Ajouter une réponse"
      v-model="answer"
      @keypress.enter="addAnswer"
    ></v-text-field>
    <v-btn color="info" @click="addAnswer">Ajouter</v-btn>

    <v-list>
      <v-list-tile
        class="panel-block"
        v-for="answer in answers"
        :key="answer.slug"
      >
        {{ answer.value }}
      </v-list-tile>
    </v-list>

    <v-btn color="info" @click="createPoll">Créer</v-btn>
  </div>
</template>

<script>
import { mapState, mapActions } from "vuex";
import randomString from "random-string";
import slugify from "slugify";
import { db } from "@/firebase";

export default {
  name: "PollCreation",
  data() {
    return {
      name: "",
      description: "",
      answer: "",
      answers: []
    };
  },
  computed: {
    ...mapState({
      userId: state => state.user.id
    })
  },
  methods: {
    ...mapActions("app", ["addNotification"]),
    async createPoll() {
      if (this.name === "" || this.answers.length < 2) {
        this.addNotification({
          text: "Veuillez renseigner l'ensemble des informations requises",
          status: "error"
        });
        return;
      }

      try {
        let users = [this.userId];

        await db.collection("polls").add({
          name: this.name,
          description: this.description,
          answers: this.answers,
          owner: this.userId,
          isActive: true,
          token: randomString({ length: 20 }),
          users
        });

        this.addNotification({
          text: "Scrutin créé mon pote",
          status: "success"
        });

        this.name = "";
        this.description = "";
        this.answer = "";
        this.answers = [];
      } catch (error) {
        this.addNotification({
          text: `Il y a eu un problème lors de la création du scrutin`,
          status: "error"
        });
      }
    },
    addAnswer() {
      if (this.answers.find(answer => answer.value === this.answer)) {
        this.addNotification({
          text: "Cette réponse existe déjà",
          status: "warning"
        });

        return;
      }

      this.answers = [
        ...this.answers,
        {
          value: this.answer,
          slug: slugify(this.answer, { lower: true, replacement: "-" })
        }
      ];
      this.answer = "";
    }
  }
};
</script>

<style></style>
