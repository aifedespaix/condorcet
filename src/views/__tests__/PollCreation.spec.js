import { shallowMount } from "@vue/test-utils";
import PollCreation from "../PollCreation.vue";

describe("PollCreation.vue", () => {
  it("should render", () => {
    const wrapper = shallowMount(PollCreation);

    expect(wrapper).toMatchSnapshot();
  });
});
