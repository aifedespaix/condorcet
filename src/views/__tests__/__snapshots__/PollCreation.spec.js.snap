// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollCreation.vue should render 1`] = `
<div>
  <v-text-field-stub label="Nom" value=""></v-text-field-stub>
  <v-textarea-stub color="primary" errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" label="Description" clearicon="$vuetify.icons.clear" type="text" rowheight="24" rows="5" name="input-7-1"></v-textarea-stub>
  <v-text-field-stub label="Ajouter une réponse" value=""></v-text-field-stub>
  <v-btn-stub color="info" activeclass="v-btn--active" tag="button" type="button">Ajouter</v-btn-stub>
  <v-list-stub></v-list-stub>
  <v-btn-stub color="info" activeclass="v-btn--active" tag="button" type="button">Créer</v-btn-stub>
</div>
`;
